src/AggregateFunctions/AggregateFunctionUniq.h

```diff
@@ -130,30 +131,20 @@ namespace detail
   */
 template <typename T> struct AggregateFunctionUniqTraits
 {
-    static UInt64 hash(T x) { return x; }
-};
-
-template <> struct AggregateFunctionUniqTraits<UInt128>
-{
-    static UInt64 hash(UInt128 x)
-    {
-        return sipHash64(x);
-    }
-};
-
-template <> struct AggregateFunctionUniqTraits<Float32>
-{
-    static UInt64 hash(Float32 x)
-    {
-        return ext::bit_cast<UInt64>(x);
-    }
-};
-
-template <> struct AggregateFunctionUniqTraits<Float64>
-{
-    static UInt64 hash(Float64 x)
+    static UInt64 hash(T x)
     {
-        return ext::bit_cast<UInt64>(x);
+        if constexpr (std::is_same_v<T, UInt128>)
+        {
+            return sipHash64(x);
+        }
+        else if constexpr (std::is_same_v<T, Float32> || std::is_same_v<T, Float64>)
+        {
+            return ext::bit_cast<UInt64>(x);
+        }
+        else if constexpr (sizeof(T) <= sizeof(UInt64))
+            return x;
+        else
+            return DefaultHash64<T>(x);
```

Comment:
It looks like we calculated wrong hash for Decimal128 before. It would be a compile time error for new types now.